cmd/bosun/sched/check.go

```diff
@@ -185,6 +186,52 @@ func (s *Schedule) RunHistory(r *RunHistory) {
 	if checkNotify && s.nc != nil {
 		s.nc <- true
 	}
+	// Instrument alert states.
+	severityCount := make(map[string]int64)
+	lastAbnormalCount := make(map[string]int64)
+	unAckedCount := map[bool]int64{true: 0, false: 0}
+	activeCount := map[bool]int64{true: 0, false: 0}
+	for _, state := range s.status {
+		if !state.Open {
+			return
+		}
+		severity := state.Status().String()
+		if _, ok := severityCount[severity]; ok {
+			severityCount[severity]++
+		} else {
+			severityCount[severity] = 1
+		}
+		lastAbnormal := state.AbnormalStatus().String()
+		if _, ok := lastAbnormalCount[lastAbnormal]; ok {
+			lastAbnormalCount[lastAbnormal]++
+		} else {
+			lastAbnormalCount[lastAbnormal] = 1
+		}
+		unAckedCount[state.NeedAck]++
+		activeCount[state.IsActive()]++
+	}
+	for k, v := range severityCount {
+		collect.Put("alerts.current_severity", opentsdb.TagSet{"severity": k}, v)
+	}
+	metadata.AddMeta("bosun.alerts.current_severity", nil, "rate", metadata.Gauge, false)
+	metadata.AddMeta("bosun.alerts.current_severity", nil, "unit", metadata.Alert, false)
+	metadata.AddMeta("bosun.alerts.current_severity", nil, "desc", "The number of open alerts by current severity.", false)
+	for k, v := range lastAbnormalCount {
+		collect.Put("alerts.last_abnormal_severity", opentsdb.TagSet{"severity": k}, v)
+	}
+	metadata.AddMeta("bosun.alerts.bosun.alerts.last_abnormal_severity", nil, "rate", metadata.Gauge, false)
+	metadata.AddMeta("bosun.alerts.bosun.alerts.last_abnormal_severity", nil, "unit", metadata.Alert, false)
+	metadata.AddMeta("bosun.alerts.bosun.alerts.last_abnormal_severity", nil, "desc", "A number of open alerts by last abnormal severity.", false)
```

Comment:
Actually in this case I don't think we should use constants since the description documents the particular instrumentation in the code (so proximity wins over neatness). That is different than the way scollector uses metadata.